docs/upgrading-to-8.x.asciidoc

```diff
@@ -0,0 +1,233 @@
+...
+----
+====
+
+. **Review your configuration file**
++
+Some settings have been removed or changed. You may need to update your `apm-server.yml` configuration
+file prior to starting the APM Server.
+See <<directory-layout>> for help in locating this file,
+and <<configuring-howto-apm-server>> for a list of all available configuration options.
+
+. **Start the APM Server**
++
+To start the APM Server, run:
++
+[source,bash]
+----
+./apm-server -e
+----
++
+Additional details are available in <<apm-server-starting,start the APM Server>>.
+
+. **(Optional) Upgrade to the APM integration**
++
+Got time for one more upgrade?
+See <<upgrade-to-apm-integration>>.
+
+// ********************************************************
+
+[[upgrade-8.0-self-integration]]
+==== Upgrade a self-installation of the APM integration to {version}
+
+++++
+<titleabbrev>Self-installation APM integration</titleabbrev>
+++++
+
+This upgrade guide is for the Elastic APM integration.
+Only use this guide if both of the following are true:
+
+* You have a self-installation of the {stack}, i.e. you're not using {ecloud}.
+* You have already upgraded to and are running {fleet} and the Elastic APM integration.
+
+[float]
+==== Prerequisites
+
+. Prior to upgrading to version {version}, {es}, and {kib}
+must be upgraded to version 7.17. To upgrade {es} and {kib},
+see the https://www.elastic.co/guide/en/elastic-stack/7.17/upgrading-elastic-stack.html[{stack} Installation and Upgrade Guide]
+
+. Review the APM <<release-notes,release notes>>, <<apm-breaking,breaking changes>>,
+and Observability {observability-guide}/whats-new.html[What's new] content.
+
+[float]
+==== Upgrade steps
+
+. Upgrade the {stack} to version {version}.
++
+The {stack} ({es} and {kib}) must be upgraded before {agent}.
+See the {stack-ref}/upgrading-elastic-stack.html[{stack} Installation and Upgrade Guide] for guidance.
+
+. Upgrade {agent} to version {version}.
+As a part of this process, the APM integration will automatically upgrade to version {version}.
++
+--
+. In {fleet}, select **Agents**.
+
+. Under **Agents**, click **Upgrade available** to see a list of agents that you can upgrade.
+
+. Choose **Upgrade agent** from the **Actions** menu next to the agent you want to upgrade.
+The **Upgrade agent** option is grayed out when an upgrade is unavailable, or
+the {kib} version is lower than the agent version.
+--
++
+For more details, or for bulk upgrade instructions, see
+{fleet-guide}/upgrade-elastic-agent.html[Upgrade {agent}]
+
+// ********************************************************
+
+[[upgrade-8.0-cloud-standalone]]
+==== Upgrade {ecloud} APM Server standalone to {version}
+
```

Comment:
How does an ESS standalone upgrade work? I expected the APM integration to be installed as a part of this upgrade, but it wasn't—yet my data was now being stored in data streams. 

I confirmed I was still in standalone mode by finding the upgrade option in the APM app.

Reply:
Which version did you test with? You need to use at least `8.0.0-rc-1`, as the `8.0.0-beta` candidate did not include the changes. When data were written to datastreams after the upgrade, it actually does sound like the apmpackage has been installed; otherwise APM Server would block the ingestion.

Reply:
I used `v8.0.0-SNAPSHOT`. I'm not sure how to check what that includes. Agreed that it sounds like the apm package was installed—that's why I was confused when I didn't couldn't find it on the policy. I will try again next week.